pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -462,7 +467,10 @@ def _get_cache(result_metric: ResultMetric, on_step: bool) -> Optional[torch.Ten
             cache = result_metric._forward_cache
         elif not on_step and result_metric.meta.on_epoch:
             if not result_metric._computed:
+                should = result_metric.meta.sync.should
+                result_metric.meta.sync.should = True
                 result_metric.compute()
+                result_metric.meta.sync.should = should
```

Comment:
why this?

Reply:
`self.should` is False by default, and no sync were done on epoch end.

```
    @property
    def __call__(self) -> Any:
        return partial(self.fn, reduce_op=self.op, group=self.group) if self.should else self.no_op
```

Need more tests.